### PR TITLE
fix(merge): include command hash into the report name

### PR DESCRIPTION
**Pull request by yury-s**
Reference https://github.com/microsoft/playwright/issues/30091

### CODE REVIEW

File: packages/playwright/src/program.ts [hunk 1]
```diff
@@ -201,6 +202,19 @@ async function runTests(args: string[], opts: { [key: string]: any }) {
   gracefullyProcessExitDoNotHang(exitCode);
 }
 
+function computeCommandHash(args: string[], opts: { [key: string]: any }) {
+  const command = {
+    args,
+    opts: {} as any
+  };
+  // Omit shard from the hash, shard number is added explicitely at the end of the blob report file name.
+  const keys = Object.keys(opts).filter(k => k !== 'shard');
```
Comment (pavelfeldman) [inline packages/playwright/src/program.ts:211]: nit: i looked at the options and we don't seem to care about most of them. So maybe only use those definding the scope of the run? `[args], { grep, grepInvert, project, shard }`

**Review by pavelfeldman — COMMENTED** [review]
How about supporting grep via env while we are there?

File: packages/playwright/src/runner/reporters.ts [hunk 2]
```diff
@@ -90,9 +91,27 @@ function reporterOptions(config: FullConfigInternal, mode: 'list' | 'test' | 'me
...
 
+function computeCommandHash(config: FullConfigInternal) {
+  let parts = [];
+  // Include project names for readability.
+  if (config.cliProjectFilter)
+    parts.push(...config.cliProjectFilter);
+  const command = {} as any;
+  if (config.cliArgs.length)
+    command.cliArgs = config.cliArgs;
+  if (config.cliGrep)
```
Comment (pavelfeldman) [inline packages/playwright/src/runner/reporters.ts:106]: This does not address the passed-via-env scenario.

**Review by pavelfeldman — APPROVED** [review]

**Comment by 1337doctor** [comment]
If your gh actions for saving or restoring shard from cache (artifacts) and your pipeline stopped working - maybe this post prevents some headache.

If you receive this error " Path Validation Error: Path(s) specified in the action for caching do(es) not exist, hence no cache is being saved." while saving or restoring a cache / artifact and your used a glob with a specific filename - this commit might be the reason.

Just refer to mybasedir/blob-report instead of mybasedir/blob-report/report-1.zip (or something similiar).

Hope someone finds this helpful.

**Pull request merged**